src/writer/direntPool.h

```diff
@@ -46,11 +46,22 @@ namespace zim
           }
         }
 
-        Dirent* getDirent() {
+        Dirent* getClassicDirent(char ns, const std::string& path, const std::string& title, uint16_t mimetype) {
           if (direntIndex == 0xFFFF) {
             allocate_new_pool();
           }
-          return pools.back() + direntIndex++;
+          auto dirent = pools.back() + direntIndex++;
+          new (dirent) Dirent(ns, path, title, mimetype);
+          return dirent;
+        }
+
+        Dirent* getRedirectDirent(char ns, const std::string& path, const std::string& title, char targetNs, const std::string& targetPath) {
+          if (direntIndex == 0xFFFF) {
+            allocate_new_pool();
+          }
+          auto dirent = pools.back() + direntIndex++;
+          new (dirent) Dirent(ns, path, title, targetNs, targetPath);
```

Comment:
Here (both in `getClassicDirent()` and `getRedirectDirent()`) you construct a new `Dirent` over a default-constructed (placeholder) `Dirent` object without destroying the latter first. With current code that is not a problem since a default-constructed `Dirent` doesn't own any resources. But if that changes in the future, a memory leak will be introduced.

Reply:
You're right.
I've made the `DirentPool` allocate the Dirent but not construct them.
We are inplace constructing the Dirent when we really need it.